abi/cbor_string.go

```diff
@@ -0,0 +1,46 @@
+package abi
+
+import (
+	"io"
+
+	"golang.org/x/xerrors"
+
+	cbg "github.com/whyrusleeping/cbor-gen"
+)
+
+type CborString string
```

Comment:
If its possible it would be better to gen this directly (updates to cbor gen propagate, no need to review independently for correctness).  Does that not work with cbg?  If so it seems like its worth filing an issue over there.

Reply:
Cbor-gen only works with structs, not primitives or type definitions thereof. There are hardcoded utilities for some primitives like `[CborInt](https://github.com/whyrusleeping/cbor-gen/blob/f29c37e9c44c4a318d70f78686893ec960cfa771/utils.go#L712)` in the library, but there doesn't seem to be one for `string`